mkosi/distributions/debian.py

```diff
@@ -190,6 +191,68 @@ def install(cls, context: Context) -> None:
 
         cls.install_packages(context, [Path(deb).name.partition("_")[0].removesuffix(".deb") for deb in essential])
 
+        if context.config.overlay or context.config.output_format in (OutputFormat.sysext, OutputFormat.confext):
+            return
+
+        # Debian being Debian means we need to special case handling os-release. Fix the content to actually
+        # match what we are building, and set up a diversion so that dpkg doesn't overwrite it on package updates.
+        for candidate in ["etc/os-release", "usr/lib/os-release", "usr/lib/initrd-release"]:
+            osrelease = context.root / candidate
+            # at this point we know we will either change or add to the file
+            newosrelease = osrelease.with_suffix(".new")
+
+            if not osrelease.is_file():
+                continue
+
+            if osrelease.is_symlink() and candidate != "etc/os-release":
+                continue
+
+            image_id_written = image_version_written = False
+            with osrelease.open("r") as old, newosrelease.open("w") as new:
+                # fix existing values
+                for line in old.readlines():
+                    if context.config.image_id and line.startswith("IMAGE_ID="):
+                        new.write(f'IMAGE_ID="{context.config.image_id}"\n')
+                        image_id_written = True
+                    elif context.config.image_version and line.startswith("IMAGE_VERSION="):
+                        new.write(f'IMAGE_VERSION="{context.config.image_version}"\n')
+                        image_version_written = True
+                    elif (
+                        context.config.release in ("unstable", "sid") and
+                        line.startswith("VERSION_CODENAME=")
+                    ):
+                        # Workaround for Debian brokeness: unstable sets VERSION_CODENAME=<testing codename>
+                        new.write('VERSION_CODENAME=sid\n')
+                    else:
+                        new.write(line)
+
+                # append if they were missing
+                if context.config.image_id and not image_id_written:
+                    new.write(f'IMAGE_ID="{context.config.image_id}"\n')
+                if context.config.image_version and not image_version_written:
+                    new.write(f'IMAGE_VERSION="{context.config.image_version}"\n')
```

Comment:
So why can't we just fix up the `VERSION_CODENAME` here and set up the divert and then leave the fixing of `IMAGE_ID` and `IMAGE_VERSION` to the generic `configure_os_release()` method? I don't see why we have to do the `IMAGE_ID` stuff here as well.